polkadot/runtime/parachains/src/scheduler.rs

```diff
@@ -356,40 +356,47 @@ impl<T: Config> Pallet<T> {
 	}
 
 	// on new session
-	fn push_claim_queue_items_to_assignment_provider() {
-		for (_, claim_queue) in ClaimQueue::<T>::take() {
-			// Push back in reverse order so that when we pop from the provider again,
-			// the entries in the claim queue are in the same order as they are right now.
-			for para_entry in claim_queue.into_iter().rev() {
-				Self::push_assignment(para_entry);
+	fn maybe_resize_claim_queue() {
+		let new_core_count = Self::num_cores();
+		let old_core_count = ClaimQueue::<T>::get().len() as u32;
```

Comment:
I believe the claim queue can have holes, thus this check seems incorrect?

Reply:
right, fixed